node/src/components/network.rs

```diff
@@ -0,0 +1,610 @@
+    }
+}
+
+impl<REv, P: Debug> Debug for Network<REv, P> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Network")
+            .field("our_id", &self.our_id)
+            .field("peers", &self.peers)
+            .field("listening_addresses", &self.listening_addresses)
+            .field("known_addresses", &self.known_addresses)
+            .finish()
+    }
+}
+
+impl<REv: ReactorEventT<P>, P: PayloadT> Component<REv> for Network<REv, P> {
+    type Event = Event<P>;
+    type ConstructionError = Infallible;
+
+    fn handle_event(
+        &mut self,
+        effect_builder: EffectBuilder<REv>,
+        _rng: &mut NodeRng,
+        event: Self::Event,
+    ) -> Effects<Self::Event> {
+        trace!(?self, "{}: {:?}", self.our_id, event);
+        match event {
+            Event::ConnectionEstablished {
+                peer_id,
+                endpoint,
+                num_established,
+            } => self.handle_connection_established(peer_id, endpoint, num_established),
+            Event::ConnectionClosed {
+                peer_id,
+                endpoint,
+                num_established,
+                cause,
+            } => self.handle_connection_closed(peer_id, endpoint, num_established, cause),
+            Event::UnreachableAddress {
+                peer_id,
+                address,
+                error,
+                attempts_remaining,
+            } => self.handle_unreachable_address(peer_id, address, error, attempts_remaining),
+            Event::UnknownPeerUnreachableAddress { address, error } => {
+                self.handle_unknown_peer_unreachable_address(effect_builder, address, error)
+            }
+            Event::NewListenAddress(address) => self.handle_new_listen_address(address),
+            Event::ExpiredListenAddress(address) => {
+                self.handle_expired_listen_address(effect_builder, address)
+            }
+            Event::ListenerClosed { addresses, reason } => {
+                self.handle_listener_closed(effect_builder, addresses, reason)
+            }
+            Event::ListenerError { error } => self.handle_listener_error(error),
+
+            Event::IncomingMessage { peer_id, msg: _ } => {
+                trace!(%peer_id, "{}: incoming message", self.our_id);
+                // self.handle_message(effect_builder, peer_id, msg)
+                Effects::new()
+            }
+            Event::NetworkRequest {
+                request:
+                    NetworkRequest::SendMessage {
+                        dest,
+                        payload: _,
+                        responder,
+                    },
+            } => {
+                // We're given a message to send out.
+                trace!(%dest, "{}: request to send message", self.our_id);
+                // self.send_message(dest, Message(payload));
+                responder.respond(()).ignore()
+            }
+            Event::NetworkRequest {
```

Comment:
We should remove broadcast now, right, or make it an alias for gossiping?

Reply:
We'll need to maintain support for broadcast until around the point we remove the small_network component.  I think until we have consensus refactored to not call broadcast, we should retain support for it here.  I think that should be pretty painless, but if I'm wrong we can revisit this approach.

Reply:
I assume that semantically they are close enough, right? Should be force the change in consensus now?

Reply:
Semantically they are close.  But if we change to use small_network gossip rather than small_network broadcast, consensus will stop working.  We'd need to change consensus to use a new vertex gossiper component.  I'd really rather avoid that in case we can replace our gossiper components altogether.

Reply:
My bad, I thought that work was already in, but it's only planned.